server/trpc/router/profile.ts

```diff
@@ -16,6 +24,43 @@ export const profileRouter = router({
       });
       return profile;
     }),
+  updateProfilePhotoUrl: protectedProcedure
+    .input(updateProfilePhotoUrlSchema)
+    .mutation(async ({ input, ctx }) => {
+      const profile = await ctx.prisma.user.update({
+        where: {
+          id: ctx.session.user.id,
+        },
+        data: {
+          image: input.url,
+        },
+      });
+      return profile;
+    }),
+  getUploadUrl: protectedProcedure
+    .input(uploadPhotoUrlSchema)
+    .mutation(async ({ ctx, input }) => {
+      const { size, type } = input;
+      const extension = type.split("/")[1];
+
+      if (!["jpg", "jpeg", "gif", "png", "webp"].includes(extension)) {
+        throw new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Invalid file format.",
+        });
+      }
+
+      if (size > 8388608) {
```

Comment:
can we keep this somewhere as a constant?

Reply:
I'll refactor this later, I think there is a more elegant way to do this with the s3 client I'm just still learning it.